### ISSUE TITLE
Methods In Java

### ISSUE THREAD

**Issue by rkpatra201** [CORE-JAVA-101/dsa-series-2023#3]
In Java, methods are used to define a block of code that performs a specific task. They allow you to reuse code by encapsulating a set of instructions that can be called multiple times throughout your program. Methods can be called by other methods or by the main program to perform a specific action, return a value, or modify an object's state.

Some of the key uses of methods in Java include:

1. Code Reusability: By defining methods, you can write code once and reuse it multiple times, avoiding code duplication.

2. Modularity: Methods break down complex problems into smaller, more manageable tasks, making it easier to understand and maintain the code.

3. Abstraction: Methods allow you to hide the complexity of an operation by exposing only the inputs and outputs required to perform a task.

4. Encapsulation: Methods can be used to define a set of operations that work together to modify an object's state, keeping the code organized and reducing the risk of errors.

5. Flexibility: Methods can be overloaded, which means you can define multiple methods with the same name but different parameter lists, allowing you to perform the same task with different types of input data.

In summary, methods in Java are a powerful programming construct that allows you to write modular, reusable, and maintainable code.

**Comment by rkpatra201** [comment]
# Method Return Types

In Java, every method has a return type, which specifies the type of data that the method will return after it has completed its execution. The return type of a method is specified in the method signature and is indicated before the method name.

Java supports several types of return types, including:

1. void: A method with a void return type does not return any value. It is typically used for methods that perform some action or have a side effect but do not need to return a value.

2. Primitive types: Methods can return any of the primitive data types in Java, such as int, double, float, boolean, and char.

3. Objects: Methods can also return objects of any class, including user-defined classes.

4. Arrays: Methods can return an array of any type, including primitive types, wrapper types, user defined types and objects.

Here are some examples of methods with different return types:

```java
public class Calculator {
  public int x;
  public int y;

  public Calculator(int i, int j) {
    this.x = i;
    this.y = j;
  }
}


public class App {
  public static void main(String[] args) {
    showMessage("Hello World"); // 1
    long time1 = getCurrentTimeAsMilliSecond(); // 2
    System.out.println("time1: " + time1);
    Long time2 = getCurrentTimeAsMilliSecond_1(); // 3
    System.out.println("time2: " + time2);
    String response = sayHello(); // 4
    System.out.println("response: " + response);
    int[] ages = getAges(); // 5
    System.out.println("ages: " + Arrays.toString(ages));
    Calculator calculator = getCalculator(); // 6
    System.out.println("calculator object: " + calculator.x);
  }

  // method with void return type
  public static void showMessage(String message) {
    System.out.println(message);
  }

  // method with primitive(long) return type
  public static long getCurrentTimeAsMilliSecond() {
    return System.currentTimeMillis();
  }

  // method with wrapper(Long) return type
  public static Long getCurrentTimeAsMilliSecond_1() {
    Long value = System.currentTimeMillis();
    return value;
  }

  // method with return type as: java predefined class (String) type
  public static String sayHello() {
    String message = "hello";
    return message;
  }

  // method return type as array
  public static int[] getAges() {
    int[] arr = {10, 20, 30, 40};
    return arr;
  }


  // method with return type as: user defined type or custom class(Calculator) type
  public static Calculator getCalculator() {
    int i = 10; // local variable i can be accessed only inside the method
    int j = 20;  // local variable j can be accessed only inside the method
    Calculator calculator = new Calculator(i, j);
    return calculator;
  }

}
````

1. called the void method `showMessage` by passing a String parameter.
2. called the method `getCurrentTimeAsMilliSecond` whose return value is primitive type long and stored the return value in the primitive type long variable #time1
3. called the method `getCurrentTimeAsMilliSecond_1` whose return value is wrapper type and stored the return value in the wrapper type Long variable #time2
4. called the method `sayHello` whose return value is String type and stored the return value in the String type variable #response
5. called the method `getAges` whose return value is int[] and stored the return value in the int[] type variable #ages.
6. called the method `getCalculator` whose return value is Calculator and stored the return value in the Calculator type variables #calculator

**Comment by rkpatra201** [comment]
# Method Params
